source/documentNavigation/paragraphHelper.py

```diff
@@ -0,0 +1,301 @@
+	acceptable = True
+	# disallow if in a Word document and not using UIA, as Word has performance issues
+	if isinstance(ti, WordDocumentTextInfo) or isinstance(ti, BrowseModeWordDocumentTextInfo):
+		acceptable = False
+	# disallow if EditableTextDisplayModelTextInfo, as has performance issues (TextPad for example)
+	if isinstance(ti, EditableTextDisplayModelTextInfo):
+		acceptable = False
+	return acceptable
+
+
+def _isLastLineOfParagraph(line: str) -> bool:
+	stripped = line.strip(' \t')
+	return stripped.endswith('\r') or stripped.endswith('\n')
+
+
+def _splitParagraphIntoChunks(paragraph: str) -> Generator[str, None, None]:
+	"""
+	This function attempts to break large paragraphs into smaller chunks
+	with the goal of improving processing efficiency by some synthesizers.
+	If this function fails to break the paragraph into chunks, functionality will likely be fine,
+	though responsiveness may not be optimal.
+	"""
+	from documentNavigation.sentenceHelper import _findEndOfSentence
+	paragraphLen = len(paragraph)
+	if paragraphLen <= PREFERRED_CHUNK_SIZE:
+		yield paragraph
+		return
+	sentenceEndPoints: List[int] = []
+	endPoint = _findEndOfSentence(paragraph, 0)
+	while endPoint is not None:
+		sentenceEndPoints.append(endPoint)
+		endPoint = _findEndOfSentence(paragraph, endPoint)
+	chunkStart = chunkEnd = startIndex = endIndex = 0
+	while endIndex < len(sentenceEndPoints):
+		chunkLength = sentenceEndPoints[endIndex] - chunkStart
+		if chunkLength > PREFERRED_CHUNK_SIZE:
+			if endIndex > startIndex:
+				chunkEnd = sentenceEndPoints[endIndex - 1]
+			else:
+				chunkEnd = sentenceEndPoints[endIndex]
+				endIndex += 1
+			yield paragraph[chunkStart: chunkEnd]
+			chunkStart = chunkEnd
+			startIndex = endIndex
+		else:
+			endIndex += 1
+	if chunkStart < paragraphLen:  # any leftovers?
+		yield paragraph[chunkStart:]
+
+
+def speakParagraph(ti: textInfos.TextInfo) -> None:
+	paragraph = ""
+	numLines = 0
+	tempTi = ti.copy()
+	while numLines < MAX_LINES:
+		tempTi.expand(textInfos.UNIT_LINE)
+		line = tempTi.text.strip()
+		if len(line) > 0:
+			paragraph += line + " "
+		if _isLastLineOfParagraph(tempTi.text):
+			break
+		if not tempTi.move(textInfos.UNIT_LINE, _Offset.NEXT_LINE):
+			break
+		numLines += 1
+
+	if len(paragraph.strip()) > 0:
+		gen = _splitParagraphIntoChunks(paragraph)
+		for chunk in gen:
+			speech.speakMessage(chunk)
+	else:
+		# Translators: This is spoken when a paragraph is considered blank.
+		speech.speakMessage(_("blank"))
+
+
+def _notFoundMessage(nextParagraph: bool):
+	if nextParagraph:
+		# Translators: this message is given when there is no next paragraph when navigating by paragraph
+		ui.message("No next paragraph")
+	else:
+		# Translators: this message is given when there is no previous paragraph when navigating by paragraph
+		ui.message("No previous paragraph")
+
+
+def _moveTextInfoToParagraph(nextParagraph: bool, ti: textInfos.TextInfo) -> bool:
```

Comment:
I think many of the functions in this module need to be renamed to match the `singleLineBreak`/`doubleLineBreak` naming standard

Reply:
@seanbudd Sure. Are we ignoring the subtlety that doubleLineBreak means two or more line breaks? It's fine with me if so, as double line break is a reasonable name for this style, and anything else may be too long and cumbersome.

Reply:
@michaelDCurran also agreed that "multi" is more appropriate if this is the case